modules/gdnative/nativearvr/arvr_interface_gdnative.cpp

```diff
@@ -0,0 +1,286 @@
+
+	return size;
+}
+
+Transform ARVRInterfaceGDNative::get_transform_for_eye(ARVRInterface::Eyes p_eye, const Transform &p_cam_transform) {
+	void *parameters[3];
+	Transform ret;
+
+	ERR_FAIL_COND_V(data == NULL, Transform());
+
+	parameters[0] = (void *)this;
+	parameters[1] = (void *)&p_eye;
+	parameters[2] = (void *)&p_cam_transform;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_return_transform_for_eye", "godot_arvr_get_transform_for_eye", data, 3, parameters, &ret);
+
+	return ret;
+}
+
+CameraMatrix ARVRInterfaceGDNative::get_projection_for_eye(ARVRInterface::Eyes p_eye, real_t p_aspect, real_t p_z_near, real_t p_z_far) {
+	void *parameters[6];
+	CameraMatrix cm;
+
+	ERR_FAIL_COND_V(data == NULL, CameraMatrix());
+
+	parameters[0] = (void *)this;
+	parameters[1] = (void *)cm.matrix;
+	parameters[2] = (void *)&p_eye;
+	parameters[3] = (void *)&p_aspect;
+	parameters[4] = (void *)&p_z_near;
+	parameters[5] = (void *)&p_z_far;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_call_return_projection_for_eye", "godot_arvr_return_projection_for_eye", data, 6, parameters, NULL);
+
+	return cm;
+}
+
+void ARVRInterfaceGDNative::commit_for_eye(ARVRInterface::Eyes p_eye, RID p_render_target, const Rect2 &p_screen_rect) {
+	void *parameters[4];
+
+	ERR_FAIL_COND(data == NULL);
+
+	parameters[0] = (void *)this;
+	parameters[1] = (void *)&p_eye;
+	parameters[2] = (void *)&p_render_target;
+	parameters[3] = (void *)&p_screen_rect;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_call_commit_for_eye", "godot_arvr_commit_for_eye", data, 4, parameters, NULL);
+}
+
+void ARVRInterfaceGDNative::process() {
+	void *parameters[1];
+
+	ERR_FAIL_COND(data == NULL);
+
+	parameters[0] = (void *)this;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_call_method", "godot_arvr_process", data, 1, parameters, NULL);
+}
+
+void ARVRInterfaceGDNative::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_gdnative_library", "library"), &ARVRInterfaceGDNative::set_gdnative_library);
+}
+
+/////////////////////////////////////////////////////////////////////////////////////
+// some helper callbacks
+
+void GDAPI godot_arvr_blit(godot_int p_eye, godot_rid *p_render_target, godot_rect2 *p_rect) {
+	// blits out our texture as is, handy for preview display of one of the eyes that is already rendered with lens distortion on an external HMD
+	ARVRInterface::Eyes eye = (ARVRInterface::Eyes)p_eye;
+	RID *render_target = (RID *)p_render_target;
+	Rect2 screen_rect = *(Rect2 *)p_rect;
+
+	if (eye == ARVRInterface::EYE_LEFT) {
+		screen_rect.size.x /= 2.0;
+	} else if (p_eye == ARVRInterface::EYE_RIGHT) {
+		screen_rect.size.x /= 2.0;
+		screen_rect.position.x += screen_rect.size.x;
+	}
+
+	VSG::rasterizer->set_current_render_target(RID());
+	VSG::rasterizer->blit_render_target_to_screen(*render_target, screen_rect, 0);
+};
+
+godot_int GDAPI godot_arvr_get_texid_for(godot_rid *p_render_target) {
```

Comment:
The dangling `for` seems weird to me, maybe `godot_arvr_get_texid` would be enough?

Reply:
This is the Obj-C guy in me. get texture id ? which one? get_texture_id_for(p_render_target), reads like get texture id for render target. So you're getting the render targets texture.

But this isn't object-C so I'll change it :)